File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectServerClientJava/Networking/src/main/java/proiectServerClientJava/network/protobuffprotocol/ProtoUtils.java
package proiectServerClientJava.network.protobuffprotocol;

import proiectServerClientJava.domain.*;

import java.util.ArrayList;
import java.util.List;

public class ProtoUtils {
    public static ClientProtobufs.AppRequest createLoginRequest(Angajat employee){
        ClientProtobufs.Angajat myEmployee = ClientProtobufs.Angajat.newBuilder().
                setId(employee.getId()).
                setName(employee.getName()).
                setUsername(employee.getUsername()).
                setPassword(employee.getPassword()).build();

        return ClientProtobufs.AppRequest.newBuilder().setType(ClientProtobufs.AppRequest.Type.Login)
                .setAngajat(myEmployee).build();
    }

    public static ClientProtobufs.AppRequest createLogoutRequest(Angajat angajat) {
        ClientProtobufs.Angajat angajatDTO = ClientProtobufs.Angajat.newBuilder().
                setId(angajat.getId()).
                setName(angajat.getName()).
                setUsername(angajat.getUsername()).
                setPassword(angajat.getPassword()).build();
        ClientProtobufs.AppRequest request = ClientProtobufs.AppRequest.newBuilder().setType(ClientProtobufs.AppRequest.Type.Logout)
                .setAngajat(angajatDTO).build();
        return request;
    }

    public static ClientProtobufs.AppRequest createAddBiletRequest(Bilet bilet) {
        ClientProtobufs.Bilet biletDTO = ClientProtobufs.Bilet.newBuilder()
                .setId(bilet.getId())
                .setCumparatorName(bilet.getCumparatorName())
                .setIdSpectacol(bilet.getIdSpectacol())
                .setNrSeats(bilet.getNrSeats()).build();
        return ClientProtobufs.AppRequest.newBuilder().setType(ClientProtobufs.AppRequest.Type.AddBilet)
                .setBilet(biletDTO).build();
    }

    public static ClientProtobufs.AppRequest createGetSpectacolRequest(Long id) {
        ClientProtobufs.AppRequest request = ClientProtobufs.AppRequest.newBuilder()
                .setType(ClientProtobufs.AppRequest.Type.GetSpectacol)
                .setIdSpectacol(id).build();
        return request;
    }

    public static ClientProtobufs.AppRequest createGetAllSpectacoleRequest() {
        return ClientProtobufs.AppRequest.newBuilder().setType(ClientProtobufs.AppRequest.Type.GetAllSpectacole).build();
    }

    public static Angajat getAngajat(ClientProtobufs.AppRequest request) {
        return new Angajat(request.getAngajat().getId(), request.getAngajat().getName(), request.getAngajat().getUsername(), request.getAngajat().getPassword());
    }

    public static Bilet getBilet(ClientProtobufs.AppRequest request) {
        return new Bilet(request.getBilet().getId(), request.getBilet().getCumparatorName(), request.getBilet().getIdSpectacol(), request.getBilet().getNrSeats());
    }

    public static Spectacol getSpectacol(ClientProtobufs.AppRequest request) {
        return new Spectacol(request.getSpectacol().getId(),
                request.getSpectacol().getArtistName(),
                request.getSpectacol().getDate(),
                request.getSpectacol().getTime(),
                request.getSpectacol().getPlace(),
                request.getSpectacol().getAvailableSeats(),
                request.getSpectacol().getSoldSeats());
    }

    public static ClientProtobufs.AppResponse createLoginResponse(Angajat angajat) {
        ClientProtobufs.Angajat angajatDTO = ClientProtobufs.Angajat.newBuilder().
                setId(angajat.getId()).
                setName(angajat.getName()).
                setUsername(angajat.getUsername()).
                setPassword(angajat.getPassword()).build();
        ClientProtobufs.AppResponse response = ClientProtobufs.AppResponse.newBuilder().setType(ClientProtobufs.AppResponse.Type.Ok)
                .setAngajat(angajatDTO).build();
        return response;
    }
    public static ClientProtobufs.AppResponse createAddBiletResponse(Bilet bilet) {
        ClientProtobufs.Bilet biletDTO = ClientProtobufs.Bilet.newBuilder().
                setId(bilet.getId())
                .setCumparatorName(bilet.getCumparatorName())
                .setIdSpectacol(bilet.getIdSpectacol())
                .setNrSeats(bilet.getNrSeats()).build();
        return ClientProtobufs.AppResponse.newBuilder().setType(ClientProtobufs.AppResponse.Type.SoldBilet)
                .setBilet(biletDTO).build();
    }

    public static ClientProtobufs.AppResponse createGetSpectacolResponse(Spectacol spectacol) {
        ClientProtobufs.Spectacol spectacolDTO = ClientProtobufs.Spectacol.newBuilder().
                setId(spectacol.getId())
                .setArtistName(spectacol.getArtistName())
                .setDate(spectacol.getDate())
                .setTime(spectacol.getTime())
                .setPlace(spectacol.getPlace())
                .setAvailableSeats(spectacol.getAvailableSeats())
                .setSoldSeats(spectacol.getSoldSeats())
                .build();
        ClientProtobufs.AppResponse response = ClientProtobufs.AppResponse.newBuilder()
                .setType(ClientProtobufs.AppResponse.Type.GetSpectacol)
                .setSpectacol(spectacolDTO).build();
        return response;
    }


    public static ClientProtobufs.AppResponse createGetAllSpectacoleResponse(Iterable<Spectacol> spectacole) {
        List<ClientProtobufs.Spectacol> spectacole2 = new ArrayList<>();
        spectacole.forEach(spectacol -> spectacole2.add(ClientProtobufs.Spectacol.newBuilder().setId(spectacol.getId()).
                setArtistName(spectacol.getArtistName()).
                setDate(spectacol.getDate()).
                setTime(spectacol.getTime()).
                setPlace(spectacol.getPlace()).
                setAvailableSeats(spectacol.getAvailableSeats()).
                setSoldSeats(spectacol.getSoldSeats()).
                build()));
        return ClientProtobufs.AppResponse.newBuilder().setType(ClientProtobufs.AppResponse.Type.GetAllSpectacole).addAllSpectacole(spectacole2).build();
    }

    public static ClientProtobufs.AppResponse createOkResponse() {
        ClientProtobufs.AppResponse response = ClientProtobufs.AppResponse.newBuilder()
                .setType(ClientProtobufs.AppResponse.Type.Ok).build();
        return response;
    }

    public static ClientProtobufs.AppResponse createErrorResponse(String text) {
        ClientProtobufs.AppResponse response = ClientProtobufs.AppResponse.newBuilder()
                .setType(ClientProtobufs.AppResponse.Type.Error)
                .setError(text).build();
        return response;
    }

    public static String getError(ClientProtobufs.AppResponse response) {
        String errorMessage = response.getError();
        return errorMessage;
    }


    public static Angajat getAngajat(ClientProtobufs.AppResponse response) {
        return new Angajat(response.getAngajat().getId(), response.getAngajat().getName(), response.getAngajat().getUsername(), response.getAngajat().getPassword());
    }

    public static Bilet getBilet(ClientProtobufs.AppResponse response) {
        return new Bilet(response.getBilet().getId(), response.getBilet().getCumparatorName(), response.getBilet().getIdSpectacol(), response.getBilet().getNrSeats());
    }

    public static Spectacol getSpectacol(ClientProtobufs.AppResponse response) {
        return new Spectacol(response.getSpectacol().getId(),
                response.getSpectacol().getArtistName(),
                response.getSpectacol().getDate(),
                response.getSpectacol().getTime(),
                response.getSpectacol().getPlace(),
                response.getSpectacol().getAvailableSeats(),
                response.getSpectacol().getSoldSeats());
    }

    public static Iterable <Spectacol> getAllSpectacole(ClientProtobufs.AppResponse response)
    {
        List <Spectacol> spectacole = new ArrayList<>();
        response.getSpectacoleList().forEach(spectacol -> spectacole.add(new Spectacol(spectacol.getId(),
                spectacol.getArtistName(), spectacol.getDate(), spectacol.getTime(),
                spectacol.getPlace(), spectacol.getAvailableSeats(), spectacol.getSoldSeats())));
        return spectacole;
    }

    public static long getIdSpectacol(ClientProtobufs.AppRequest request)
    {
        return request.getIdSpectacol();
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/protobuf/ProtoUtils.cs
using Chat.Protocol;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using proto = Chat.Protocol;
using System.Net.Sockets;
using proiectSCcsharp.domain;

namespace protobuf
{
    internal class ProtoUtils
    {
        public static AppRequest createLoginRequest(proiectSCcsharp.domain.Angajat employee)
        {
            var myEmployee = new proto.Angajat
            {
                Id = employee.id,
                Name = employee.name,
                Username = employee.username,
                Password = employee.password
            };
            return new AppRequest
            {
                Type = AppRequest.Types.Type.Login,
                Angajat = myEmployee
            };
        }

        public static AppRequest createLogoutRequest(proiectSCcsharp.domain.Angajat employee)
        {
            var myEmployee = new proto.Angajat
            {
                Id = employee.id,
                Name = employee.name,
                Username = employee.username,
                Password = employee.password
            };
            return new AppRequest
            {
                Type = AppRequest.Types.Type.Logout,
                Angajat = myEmployee
            };
        }
        public static AppRequest createAddBiletRequest(proiectSCcsharp.domain.Bilet bilet)
        {
            proto.Bilet myTicket = new proto.Bilet
            {
                Id = bilet.id,
                IdSpectacol = bilet.idSpectacol,
                CumparatorName = bilet.cumparatorName,
                NrSeats = bilet.nrSeats
            };
            return new AppRequest
            {
                Type = AppRequest.Types.Type.AddBilet,
                Bilet = myTicket
            };
        }


        public static AppRequest createGetSpectacolRequest(long id)
        {
            return new AppRequest
            {
                Type = AppRequest.Types.Type.GetSpectacol,
                IdSpectacol = id
            };
        }

        public static AppRequest createGetAllSpectacoleRequest()
        {
            return new AppRequest
            {
                Type = AppRequest.Types.Type.GetAllSpectacole
            };
        }

        public static proiectSCcsharp.domain.Angajat GetAngajat(AppRequest request)
        {
            proiectSCcsharp.domain.Angajat employee = new proiectSCcsharp.domain.Angajat
            (
                request.Angajat.Id,
                request.Angajat.Name,
                request.Angajat.Username,
                request.Angajat.Password
            );
            return employee;
        }

        public static proiectSCcsharp.domain.Bilet GetBilet(AppRequest request)
        {
            proiectSCcsharp.domain.Bilet ticket = new proiectSCcsharp.domain.Bilet
            (
                request.Bilet.Id,
                request.Bilet.CumparatorName,
                request.Bilet.IdSpectacol,
                request.Bilet.NrSeats
            ) ;
            return ticket;
        }
        
        public static proiectSCcsharp.domain.Spectacol GetSpectacol(AppRequest request)
        {
            proiectSCcsharp.domain.Spectacol spectacol = new proiectSCcsharp.domain.Spectacol
                (
                    request.Spectacol.Id,
                    request.Spectacol.ArtistName,
                    request.Spectacol.Date,
                    request.Spectacol.Time,
                    request.Spectacol.Place,
                    request.Spectacol.AvailableSeats,
                    request.Spectacol.SoldSeats
                );
            return spectacol;
        }

        public static AppResponse createLoginResponse(proiectSCcsharp.domain.Angajat employee)
        {
            var myEmployee = new proto.Angajat
            {
                Id = employee.id,
                Name = employee.name,
                Username = employee.username,
                Password = employee.password
            };
            return new AppResponse { Type = AppResponse.Types.Type.Ok, Angajat = myEmployee };
        }

        public static AppResponse createAddBiletResponse(proiectSCcsharp.domain.Bilet ticket)
        {
            proto.Bilet myTicket = new proto.Bilet
            {
                Id = ticket.id,
                CumparatorName = ticket.cumparatorName,
                IdSpectacol = ticket.idSpectacol,
                NrSeats = ticket.nrSeats
            };
            return new AppResponse { Type = AppResponse.Types.Type.SoldBilet, Bilet = myTicket };
        }

        public static AppResponse createGetSpectacolResponse(proiectSCcsharp.domain.Spectacol spectacol)
        {
            var mySpectacol = new proto.Spectacol
            {
                Id = spectacol.id,
                ArtistName = spectacol.artistName,
                Date = spectacol.date,
                Time = spectacol.time,
                Place = spectacol.place,
                AvailableSeats = spectacol.availableSeats,
                SoldSeats = spectacol.soldSeats
            };
            return new AppResponse { Type = AppResponse.Types.Type.GetSpectacol, Spectacol = mySpectacol };
        }
        

        public static AppResponse createGetAllSpectacoleResponse(IEnumerable<proiectSCcsharp.domain.Spectacol> shows)
        {
            var showDTOList = new List<proto.Spectacol>();
            foreach (var showDTO in shows)
            {
                showDTOList.Add(new proto.Spectacol
                {
                    Id = showDTO.id,
                    ArtistName = showDTO.artistName,
                    Date = showDTO.date,
                    Time = showDTO.time,
                    Place = showDTO.place,
                    AvailableSeats = showDTO.availableSeats,
                    SoldSeats = showDTO.soldSeats,
                });
            }
            return new AppResponse { Type = AppResponse.Types.Type.GetAllSpectacole, Spectacole = { showDTOList } };
        }

        public static AppResponse createErrorResponse(string message)
        {
            return new AppResponse { Type = AppResponse.Types.Type.Error, Error = message };
        }

        public static AppResponse createOkResponse()
        {
            return new AppResponse { Type = AppResponse.Types.Type.Ok };
        }

        public static proiectSCcsharp.domain.Angajat GetAngajat(AppResponse response)
        {
            proiectSCcsharp.domain.Angajat employee = new proiectSCcsharp.domain.Angajat
            (
                response.Angajat.Id,
                response.Angajat.Name,
                response.Angajat.Username,
                response.Angajat.Password
            );
            return employee;
        }

        public static proiectSCcsharp.domain.Bilet GetBilet(AppResponse response)
        {
            proiectSCcsharp.domain.Bilet ticket = new proiectSCcsharp.domain.Bilet
            (
                response.Bilet.Id,
                response.Bilet.CumparatorName,
                response.Bilet.IdSpectacol,
                response.Bilet.NrSeats
            );
            return ticket;
        }

        public static proiectSCcsharp.domain.Spectacol GetSpectacol(AppResponse response)
        {
            proiectSCcsharp.domain.Spectacol spectacol = new proiectSCcsharp.domain.Spectacol
                (
                    response.Spectacol.Id,
                    response.Spectacol.ArtistName,
                    response.Spectacol.Date,
                    response.Spectacol.Time,
                    response.Spectacol.Place,
                    response.Spectacol.AvailableSeats,
                    response.Spectacol.SoldSeats
                );
            return spectacol;
        }

        public static List<proiectSCcsharp.domain.Spectacol> getAllSpectacole(AppResponse response)
        {
            List<proiectSCcsharp.domain.Spectacol> showDTOList = new List<proiectSCcsharp.domain.Spectacol>();
            foreach (var showDTO in response.Spectacole)
            {
                showDTOList.Add(new proiectSCcsharp.domain.Spectacol
                (
                    showDTO.Id,
                    showDTO.ArtistName,
                    showDTO.Date,
                    showDTO.Time,
                    showDTO.Place,
                    showDTO.AvailableSeats,
                    showDTO.SoldSeats
                ));
            }
            return showDTOList;
        }

        public static long getIdSpectacol(AppRequest request)
        {
            return request.IdSpectacol;
        }

        public static string getError(AppResponse response)
        {
            return response.Error;
        }

    }
}
